### ISSUE TITLE
kafka source: document client.id assignment

### ISSUE THREAD

**Issue by candlerb** [vectordotdev/vector#9344]
The [kafka source](https://vector.dev/docs/reference/configuration/sources/kafka/) documentation doesn't say how (or if) the `client.id` is assigned.

Through checking the [source](https://github.com/vectordotdev/vector/blob/v0.16.1/src/sources/kafka.rs#L272-L287), I believe that:

* The client.id defaults to `vector`
* You can override it using:

    ```
    [sources.my_kafka_source.librdkafka_options]
      "client.id" = "blah"
    ```

(aside: maybe worth giving its own setting?)

**Comment by jszwedko** [comment]
Yeah, agreed, if we are defaulting it, we should expose it as a top-level config option. Or we should document all of the rdkafka options on https://vector.dev/docs/reference/configuration/sources/kafka/ (including the defaults).

**Comment by jbvmio** [comment]
Any way the client.id can default to the underlying hostname instead of being hardset? Perhaps fallback to using "vector" if that fails? We deploy a lot of consumers on Kubernetes and having this set by default would be very beneficial.

**Comment by jszwedko** [comment]
Hi @jbvmio !

There isn't a way to do that dynamically in Vector currently, but you may be able to leverage the `HOSTNAME` environment variable that is commonly set by OSes to do something like:

```toml
[sources.my_kafka_source.librdkafka_options]
  "client.id" = "${HOSTNAME}"
```
